qa/tasks/cephfs/test_damage.py

```diff
@@ -180,8 +180,8 @@ def __hash__(self):
                 ))
 
         # Truncations
-        for obj_id in data_objects:
-            if obj_id == "500.00000000":
+        for o in data_objects:
+            if o == "500.00000000":
```

Comment:
I don't understand why "mutation for Truncate is done on obj_id of the last iteration of the previous loop". python bug?

Reply:
The value for `o` comes from this loop: https://github.com/ceph/ceph/blob/master/qa/tasks/cephfs/test_damage.py#L101

Reply:
It's not a Python bug; it's just dumb language design.

Reply:
The problem is that Python has function-level scoping. Variables are not block scoped. This means if you define a local anywhere in the function, it is visible everywhere else.

Reply:
I don't get this one either. If both ```o``` and ```obj_id``` are already declared elsewhere in the function, what bug is present and why does reusing the ```o``` signifier instead of the ```obj_id``` one fix it?

Reply:
The actual bug is here: https://github.com/ceph/ceph/pull/25843/files#diff-12a67c6bd0df8aa10d59e9e9846b8bc6R193

The variable passed to `MetadataMutation`, `o`, is not the iterator of this for loop but a previous loop.

My fix for this is to use `o` in all of the loops.

Reply:
👍